### PR TITLE
volmods

### PR DESCRIPTION
**Pull request by rocky4570**
Moto is cool! Added changes to show attached volumes on an instance (handling ghost volume as well), search volumes/snapshots by their IDs and show placement and start/create times based on API usage.

### CODE REVIEW

File: moto/ec2/models.py [hunk 1]
```diff
@@ -1444,6 +1457,13 @@ def attach_volume(self, volume_id, instance_id, device_path):
             return False
 
         volume.attachment = VolumeAttachment(volume, instance, device_path)
+        # Modify instance to capture mount of block device.
+        bdt = BlockDeviceType(volume_id=volume_id, status=volume.status, size=volume.size,
+                              attach_time=utc_date_and_time())
+        try:
+            instance.block_device_mapping[device_path] = bdt
+        except:
```
Comment (spulec) [inline moto/ec2/models.py]: When does the exception happen? Can we specify the Exception type?

**Comment by spulec** [comment]
This is great! I just had one comment/question and then happy to get this merged.

File: moto/ec2/models.py (hunk 1 above)
Comment (rocky4570) [inline moto/ec2/models.py]: The setdefault doesn't clear out the /dev/sda1 key which is introduced on a launch instance in moto (i call it a ghost volume - as it doesn't exist when describe_all_volumes() but shows up when looking at block_device_mapping of the instance - initially was going to deleted but decided to keep for backward compatability). So I just used a try/except to get around it, but could have dealt with it specifically for /dev/sda1 I guess.

**Comment by spulec** [comment]
Thanks!

**Pull request merged**